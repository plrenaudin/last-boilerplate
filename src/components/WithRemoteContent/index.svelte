<script>
  import { onMount } from "svelte";

  let loadingPromise;
  onMount(() => {
    loadingPromise = getTime();
  });

  const getTime = async () => {
    const response = await fetch("/api/time");
    const time = await response.text();
    if (response.ok) {
      return time;
    } else {
      throw new Error(time);
    }
  };
</script>

{#await loadingPromise}
  <p>....waiting</p>
{:then time}
  <p>The time is {time}</p>
{:catch error}
  <p class="text-red-600">Failed to retrieve time from server</p>
{/await}
