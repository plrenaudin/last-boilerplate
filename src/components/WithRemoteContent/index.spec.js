import WithRemoteContent from "./index.svelte";
import { render } from "@testing-library/svelte";

describe("My Component", () => {
  it("renders My Component", () => {
    const { container } = render(WithRemoteContent);

    expect(container).toMatchSnapshot();
  });
});
