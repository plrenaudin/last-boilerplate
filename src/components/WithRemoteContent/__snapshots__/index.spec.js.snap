// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component renders My Component 1`] = `
<body>
  <div>
    <p>
      ....waiting
    </p>
    
  </div>
</body>
`;
